request/inspector.go

```diff
@@ -7,6 +7,17 @@ import (
 )
 
 type ResponseInspector interface {
+	// InspectResponse is passed a response to inspect.
+	//
+	// An inspector must not modify the response. Doing so could impact later
```

Comment:
I don't think this is required. It is how we define the inspector and since we don't require this right now, I wouldn't make this claim. Just my two cents, though.

Reply:
This is just an attempt at documenting what users of this interface can expect.

Since there's no known ordering, any inspector that modifies the Response, could have an impact on later inspectors. As a result, warning folks to _not_ modify the Response seems like the safe path.

The comment aims to make it clear to someone writing their own ResponseInspector, that they probably shouldn't modify the Response, and if they do, they should be extra careful to do so in a way that doesn't interfere with other response inspectors.